AdSecGH/Components/3_Section/FlattenRebar.cs

```diff
@@ -1,130 +1,82 @@
 ﻿using System;
-using System.Collections.Generic;
+using System.Drawing;
 using System.Linq;
 
-using AdSecGH.Helpers;
-using AdSecGH.Helpers.GH;
+using AdSecCore;
+using AdSecCore.Functions;
+
 using AdSecGH.Parameters;
+using AdSecGH.Properties;
 
 using Grasshopper.Kernel;
 
-using Oasys.AdSec;
-using Oasys.AdSec.Reinforcement.Groups;
-using Oasys.AdSec.Reinforcement.Preloads;
-using Oasys.Profiles;
+using Oasys.GH.Helpers;
 
 using OasysGH;
-using OasysGH.Components;
-using OasysGH.Parameters;
 using OasysGH.Units;
 
-using OasysUnits;
+using AdSecSectionParameter = Oasys.GH.Helpers.AdSecSectionParameter;
+using Attribute = AdSecCore.Functions.Attribute;
 
 namespace AdSecGH.Components {
-  public class FlattenRebar : GH_OasysComponent {
-    public override Guid ComponentGuid => new Guid("879ecac6-464d-4286-9113-c15fcf03e4e6");
-    public override GH_Exposure Exposure => GH_Exposure.tertiary | GH_Exposure.obscure;
-    public override OasysPluginInfo PluginInfo => AdSecGH.PluginInfo.Instance;
-    protected override System.Drawing.Bitmap Icon => Properties.Resources.FlattenRebar;
-
-    public FlattenRebar() : base(
-      "FlattenRebars",
-      "FRb",
-      "Flatten all rebars in a section into single bars.",
-      CategoryName.Name(),
-      SubCategoryName.Cat4()) {
-      Hidden = true;
+
+  public class FlattenRebarGh : FlattenRebarFunction {
+
+    public FlattenRebarGh() {
+      AdSecSection.OnValueChanged += goo => {
```

Comment:
I have not run the code, but how is this event useful here? Can we not simply use
if (AdSecSection.Value != null) {
  Section.Value = AdSecSection.Value.Value.Section;
}